### ISSUE TITLE
torch.trapz with empty dim

### ISSUE THREAD

**Issue by ailzhang** [pytorch/xla#893]
```
import torch
import torch_xla

device = 'xla:0'
# device='cpu'

sizes = (0, 2)
dim=1
x=[1.0, 2.0]

t = torch.randn(sizes, device=device)

a = torch.trapz(t, x=torch.tensor(x, device=device), dim=dim)
print(a)
```
Expected: should match cpu result.
